### DISCUSSION TITLE
Python Stub Files from C++ docs (*.pyi)

### DISCUSSION THREAD

**Discussion by jasondurbin** [thliebig/openEMS-Project#183] [Ideas]
Hi,

Recently decided to jump into using openEMS after using CST and HFSS for over a decade. I'm very excited to try this out and some of the examples I have tried are very straightforward and easy to follow. 

I'm a heavy Python user: Windows + VSCode. I've used Octave in the past but I prefer Python + matplotlib + numpy. One thing that is extremely useful for Python development in VSCode are Python stub files (*.pyi). When documentation/type hinting cannot come from a source Python file, VSCode will look for *.pyi files. Has someone already created *.pyi files that I have missed?

It seems like it would be simple to create them from the docs... but I am also confused which repo the docs (https://docs.openems.de) are coming from.

**Comment by biergaizi** [comment]
Documentation is generated from `thliebig/openEMS-Project/doc-src`. The directory itself contains most of the documentation, but the API documentation for CSXCAD and openEMS are included from external sources via two symbolic links to `thliebig/CSXCAD/python/doc` and `thliebig/openEMS/python/doc`, respectively.

Study Pull Request #77 if you want to contribute to documentation.
